src/java_bytecode/java_object_factory.cpp

```diff
@@ -1028,6 +1030,22 @@ void java_object_factoryt::gen_nondet_struct_init(
         substruct_in_place);
     }
   }
+
+  // If <class_identifier>.cproverValidate() can be found in the
+  // symbol table, we add a call:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // expr.cproverValidate();
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  const irep_idt function_name =
+    "java::" + as_string(class_identifier) + ".cproverValidate:()V";
+  code_function_callt fun_call;
+  fun_call.function() = symbol_exprt(function_name);
+  fun_call.arguments().push_back(address_of_exprt(expr));
+  code_typet fun_type;
+  fun_type.return_type() = void_type();
+  fun_call.function().type() = fun_type;
+  assignments.add(fun_call);
```

Comment:
Instead of unconditionally inserting the call, wouldn't it be more efficient for the SAT solver if you only insert the call only if the symbol table says that the class originally had a method `cproverValidate` with the right signature?

I mean, the C++ comment above says that this is conditional, but I see no `if` in your code?

Reply:
I corrected that to look in the symbol table before adding the call